internal/stepscfg/stepscfg.go

```diff
@@ -0,0 +1,87 @@
+	HwInfoCommandArgs            = "run,--rm,--privileged,--quiet,--net=host,-v,/var/log:/var/log,quay.io/oamizur/hardware_info,/usr/bin/hardware_info"
+	ConnectivityCheckCommand     = "podman"
+	ConnectivityCheckCommandArgs = "run,--rm,--privileged,--quiet,--net=host,-v,/var/log:/var/log,quay.io/oamizur/connectivity_check,/usr/bin/connectivity_check"
+	ExecuteCommand               = "bash"
+	ExecuteCommandArgs           = ""
+)
+
+var hwInfoStep = createStep(models.StepTypeHardwareInfo, HwInfoCommand, strings.Split(HwInfoCommandArgs, ","))
+var connectivityCheckStep = createStep(models.StepTypeConnectivityCheck, ConnectivityCheckCommand, strings.Split(ConnectivityCheckCommandArgs, ","))
+var executeStep = createStep(models.StepTypeExecute, ExecuteCommand, strings.Split(ExecuteCommandArgs, ","))
+
+var steps = map[models.StepType]models.Step{
+	models.StepTypeHardwareInfo:      hwInfoStep,
+	models.StepTypeConnectivityCheck: connectivityCheckStep,
+	models.StepTypeExecute:           executeStep}
+
+var stateSteps = make(map[string][]models.Step)
+
+type StepsConfig struct {
+}
+
+func InitiliazeSteps() {
+
+	stateSteps[common.HostStatusKnown] = append(stateSteps[common.HostStatusKnown], steps[models.StepTypeConnectivityCheck])
+
+	stateSteps[common.HostStatusInsufficient] = append(stateSteps[common.HostStatusInsufficient], steps[models.StepTypeConnectivityCheck])
+	stateSteps[common.HostStatusDisconnected] = append(stateSteps[common.HostStatusDisconnected], steps[models.StepTypeConnectivityCheck])
+
+	stateSteps[common.HostStatusDiscovering] = append(stateSteps[common.HostStatusDiscovering], steps[models.StepTypeHardwareInfo])
+	stateSteps[common.HostStatusDiscovering] = append(stateSteps[common.HostStatusDiscovering], steps[models.StepTypeConnectivityCheck])
+
+}
+
+func GetNextSteps(ClusterID strfmt.UUID, HostID strfmt.UUID, HostStatus string, log logrus.FieldLogger) models.Steps {
+
+	log.Infof("GetNextSteps cluster: ,.%s host: <%s>, host status: <%s>", ClusterID, HostID, HostStatus)
+
+	var returnSteps = models.Steps{}
+
+	if statesMap, ok := stateSteps[HostStatus]; ok {
+		//need to add the step id
+		for _, step := range statesMap {
+			var copyStep = step
+			copyStep.StepID = createStepID(copyStep.StepType)
+			returnSteps = append(returnSteps, &copyStep)
+		}
+	}
+	logSteps(returnSteps, ClusterID, HostID, log)
+	return returnSteps
+}
+
+func createStep(stepType models.StepType, stepCmd string, stepArg []string) models.Step {
+	step := &models.Step{}
+	step.StepType = stepType
+	step.Command = stepCmd
+	step.Args = stepArg
+	return *step
+}
+
+func createStepID(stepType models.StepType) string {
+	return fmt.Sprintf("%s-%s", stepType, uuid.New().String()[:8])
+}
+
+func logSteps(steps models.Steps, clusterId strfmt.UUID, hostId strfmt.UUID, log logrus.FieldLogger) {
+	if len(steps) == 0 {
+		log.Infof("No steps required for cluster <%s> host <%s>", clusterId, hostId)
+	}
+	for i := 0; i < len(steps); i++ {
```

Comment:
```suggestion
	for _, step := range steps {
```
then the log will be using step

Reply:
fixed